platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/creation/tipset/TipsetEventCreator.java

```diff
@@ -275,7 +287,9 @@ private GossipEvent createEventByOptimizingAdvancementWeight() {
             // If there are no available other parents, it is only legal to create a new event if we are
             // creating a genesis event. In order to create a genesis event, we must have never created
             // an event before and the current non-ancient event window must have never been advanced.
-            if (nonAncientEventWindow != NonAncientEventWindow.INITIAL_EVENT_WINDOW || lastSelfEvent != null) {
+            if (nonAncientEventWindow != NonAncientEventWindow.INITIAL_EVENT_WINDOW_GENERATION
```

Comment:
Would it make sense to create a method `NonAncientEventWindow::isGenesis`?

Reply:
created